Sources/AsyncDNSResolver/c-ares/DNSResolver_c-ares.swift

```diff
@@ -276,11 +280,6 @@ extension Ares {
             }
```

Comment:
Can we also change `struct QueryReplyHandler` to `class QueryReplyHandler` please? That combined with `pointer.deinitialize(count: 1)` I see `deinit` getting called.

Reply:
Absolutely, just pushed a new commit to reflect those changes.

I'm interested in understanding the rationale behind class vs struct here. Could you please share more about your thought process? Swift noob eager to learn here 😄 

My best guess is that class deinit can free up its members, though my tests indicate no leaks with struct.

Reply:
I was told that the address of a struct is not stable, plus we are doing all these pointer manipulations already so class probably makes more sense.